server/openapi_server/controllers/deidentified_notes_controller.py

```diff
@@ -1,72 +1,70 @@
-            person_names = response.json()
-
-        # Create deidentified notes
-        for note in notes:
-            note_id = note['id']
-            res.append(deidentify_note(note,
-                [d for d in dates if d['noteId'] == note_id],
-                [d for d in person_names if d['noteId'] == note_id]))
-
-    return jsonify(res)
-
-
-def deidentify_note(note, dates, person_names):
-    """
-    Returns the deidentified clinical note where annotations are masked with '*'.
-    """
-    mask_character = '*'
-    text = note['text']
-
-    for annotation in dates:
-        mask = mask_character * annotation['length']
-        text = text[:annotation['start']] + mask + text[annotation['start'] + annotation['length']:]
-
-    for annotation in person_names:
-        mask = mask_character * annotation['length']
-        text = text[:annotation['start']] + mask + text[annotation['start'] + annotation['length']:]
-
-    note['text'] = text
-    return note
\ No newline at end of file
+        deid_request = DeidentifyRequest.from_dict(connexion.request.get_json())
+        note = deid_request.note
+
+        # Annotations is a dict[key: list[str]]
+        annotations = {}
+
+        for annotation_type in ['text_date', 'text_person_name', 'text_physical_address']:
+            annotations[annotation_type] = annotator_client.get_annotations(note, annotation_type)
+
+        # De-identify note
+        deidentified_note = Note.from_dict(note.to_dict())
+        deidentified_annotations = {key: value.copy() for key, value in annotations.items()}
+
+        deid_config: DeidentificationConfig
+        for deid_config in deid_request.deidentification_configurations:
+            if deid_config.deidentification_strategy.masking_char_config is not None:
+                masking_char = deid_config.deidentification_strategy.masking_char_config.masking_char
+                deidentified_note, deidentified_annotations = apply_masking_char(
+                    deidentified_note,
+                    deidentified_annotations,
+                    deid_config.annotation_types,
+                    masking_char
+                )
+            elif deid_config.deidentification_strategy.redact_config is not None:
+                deidentified_note, deidentified_annotations = apply_redaction(
+                    deidentified_note,
+                    deidentified_annotations,
+                    deid_config.annotation_types
+                )
+            elif deid_config.deidentification_strategy.annotation_type_config is not None:
+                deidentified_note, deidentified_annotations = apply_annotation_type(
+                    deidentified_note,
+                    deidentified_annotations,
+                    deid_config.annotation_types
+                )
+            else:
+                return "No supported de-identification method supported in request: '%s'" % (str(deid_config.to_dict()),), 400
```

Comment:
Need to return an `Error` object. Here is the list of https://github.com/nlpsandbox/nlpsandbox-schemas/tree/develop/openapi/commons/components/responses. If none are suitable, please open a ticket/PR to add a new response that will then be referenced by the deidentifier OpenAPI spec.